internal/bindings/waf_dl.go

```diff
@@ -40,22 +40,22 @@ type wafSymbols struct {
 	run            uintptr
 }
 
-// newWafDl loads the libddwaf shared library and resolves all tge relevant symbols.
+// NewWafDl loads the libddwaf shared library and resolves all tge relevant symbols.
 // The caller is responsible for calling wafDl.Close on the returned object once they
 // are done with it so that associated resources can be released.
 func NewWafDl() (dl *WafDl, err error) {
-	file, err := lib.DumpEmbeddedWAF()
+	path, closer, err := lib.DumpEmbeddedWAF()
 	if err != nil {
 		return
```

Comment:
Should we not ignore this `err`?

```suggestion
		return nil, fmt.Errorf("write an embedded WAF library: %w", err)
```

Reply:
Ah sh*t. Should have disabled auto-merge when I saw your comments. Don't worry I will do it in another PR